internal/services/consumption/schema.go

```diff
@@ -297,3 +298,177 @@ func SchemaConsumptionBudgetCommonResource() map[string]*pluginsdk.Schema {
 		},
 	}
 }
+
+func SchemaConsumptionBudgetManagementGroupResource() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ConsumptionBudgetName(),
+		},
+
+		"management_group_name": {
```

Comment:
Could we change this to
```suggestion
		"management_group_id": {
```